src/testing/components.ts

```diff
@@ -0,0 +1,103 @@
+import { Component, Type, DebugElement, NO_ERRORS_SCHEMA, Pipe, PipeTransform } from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { TestBed, ComponentFixture } from '@angular/core/testing';
+
+import { CmsService } from '../app/core/cms/cms.service';
+import { MockCmsService } from './mock.cms.service';
+import { StubRouterLinkDirective } from './stub.routerlink.directive';
+
+interface ComponentTestCallback {
+  (fix: ComponentFixture<any>, el: DebugElement, comp: any, done?: Function): any;
+}
+
+// component test-it up!
+export function cit(desc: string, testFn: ComponentTestCallback) {
+  return it(desc, function(done) {
+    TestBed.configureTestingModule({
```

Comment:
This is the actual angular test-setup code (similar to how a module is configured).